### PR TITLE
New SPAM Cartridge + cartridge spam tweak

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request

Adds a new cartridge.
Tweaks cartridge spam system.

## Why It's Good For The Game
new feature good

Captain: The standard one.
Laywer: They're already obnoxious than VIP. gives 30 seconds nurf.
Curator: Nobody actually reads a newsfeed. Let's allow them to spam in a long cooldown time than lawyers'.
VIP: They actually don't onboard a station well, but they are so special. 30 seconds buff.
FACEBUCKS: Stealing PROVE/Value-PAK cartridge isn't a good idea in general, and HoP would not want to give them to a person in general. This one is quite lesser and you'll hardly see their spam. 3 cartridges are available from HoP cartridge vendor.  It's questionable why it's more expensive than Value-PAK.
Barber/Stage Magician: They actually can't RP well because their office room is somewhere in maints. Let's allow them to advertise something.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.
image

When you change cartridges with different cooldown time. PDA remembers when messages were sent.

--------------
image

</details>

## Changelog
:cl:
refactor: spam_enabled variable is spam_delay, 1 means 1 minute, 2 means 2 minutes.
refactor: send_message proc in PDA.dm is now send_pda_message to prevent any confusion from TGUI send_message.
add: Spam cartridges have now delay multiplyer based on its type.
add: Value-PAK cartridge (Captain) multiply is *2. (2 minutes)
add: P.R.O.V.E cartridge (Lawyer) multiply is *2.5. (2 min 30 seconds)
add: Lib-Tweet (Curator) now can spam, and its multiply is *3.5
add: TWIT cartridge (VIP) delay multiply is *1.5
add: FACEBUCKS cartridge (generic) is newly added, and its multiply is *5. can be purchased from HoP cartridge vendor with 100 credits.
add: Send all message function tells you how much time you should wait to send another.
add: Unlicensed PDA (Barber, Stage Magician) is now loaded with FACEBUCKS cartridge. Barber and Stage Magician can advertise themselves.
/:cl:

### CODE REVIEW

**Comment by DerpiestBruh** [comment]
this is so awesome

**Comment by EvilDragonfiend** [comment]
btw I can't imagine any better name than FACEBUCKS, and even I am not sure if it matches the theme.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -676,11 +676,11 @@ GLOBAL_LIST_EMPTY(PDAs)
 		t = Gibberish(t, TRUE)
 	return t
 
-/obj/item/pda/proc/send_message(mob/living/user, list/obj/item/pda/targets, everyone)
+/obj/item/pda/proc/send_pda_message(mob/living/user, list/obj/item/pda/targets, everyone, var/multi_delay=0)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/PDA/PDA.dm]:
```suggestion
/obj/item/pda/proc/send_pda_message(mob/living/user, list/obj/item/pda/targets, everyone, multi_delay=0)
```

Comment (PowerfulBacon) [reply code/game/objects/items/devices/PDA/PDA.dm]: This also shouldn't default to 0, allowing free infinite spam. It should either have no default implying that it is required, or be set to 1 if it is not a required parameter.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -783,14 +783,14 @@ GLOBAL_LIST_EMPTY(PDAs)
 	update_icon()
 	add_overlay(icon_alert)
 
-/obj/item/pda/proc/send_to_all(mob/living/U)
-	if (last_everyone && world.time < last_everyone + PDA_SPAM_DELAY)
-		to_chat(U,"<span class='warning'>Send To All function is still on cooldown.")
+/obj/item/pda/proc/send_to_all(mob/living/U, var/multi_delay)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/PDA/PDA.dm]:
```suggestion
/obj/item/pda/proc/send_to_all(mob/living/U, multi_delay)
```

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/game/objects/items/devices/PDA/PDA.dm]:
```
			if("MessageAll")
				if(cartridge?.spam_delay)
					send_to_all(U, cartridge?.spam_delay)
```

This prevents it. but I can put another cheat-proof then.

the reason why it should be given to 0 at the default parameter is that the proc is a standard sending feature too. `send_pda_message` is used in general. `multi_delay` with not-0 means that you called the proc through `send_all` proc.

Comment (EvilDragonfiend) [reply code/game/objects/items/devices/PDA/PDA.dm]:
```
	if(multi_delay)
		send_pda_message(U,get_viewable_pdas(), TRUE, multi_delay)
```

a cheat proof one - newly added.

Comment (EvilDragonfiend) [reply code/game/objects/items/devices/PDA/PDA.dm]: the reason why it should be given to 0 at the parameter is that the proc is a standard sending feature too. `send_pda_message` is used in general.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**